### PR TITLE
Tailwindui css layout support - colon is not accepted by the pattern

### LINKED ISSUE #859: [Issue] Tailwindui css layout support - colon is not accepted by the pattern
This issue is automatically created based on existing pull request: magento/magento2-page-builder#791: Tailwindui css layout support - colon is not accepted by the pattern

---------
Ashampoo_Snap_Mittwoch, 10  November 2021_09h34m59s_002_

https://tailwindcss.com/docs/responsive-design#overview
https://github.com/magento/magento2/pull/34559

### PR DESCRIPTION
**Pull request by in-session**
Ashampoo_Snap_Mittwoch, 10  November 2021_09h34m59s_002_

https://tailwindcss.com/docs/responsive-design#overview
https://github.com/magento/magento2/pull/34559

### Resolved issues:
1. [x] resolves magento/magento2-page-builder#859: Tailwindui css layout support - colon is not accepted by the pattern

### CODE REVIEW

**Review by Den4ik — APPROVED** [review]

**Review by ihor-sviziev — APPROVED** [review]

**Comment by paras89** [comment]
@magento run all tests

**Comment by paras89** [comment]
@in-session can you pull develop branch into your fork's branch and re-run the tests? After that once the test results are out I can help you fix them!

**Comment by in-session** [comment]
@magento run all tests

**Comment by in-session** [comment]
@magento run Functional Tests B2B Functional Tests EE

**Comment by in-session** [comment]
https://github.com/magento run Functional Tests B2B, Functional Tests EE

**Comment by in-session** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by paras89** [comment]
@magento run Functional Tests B2B

**Comment by in-session** [comment]
@magento run Functional Tests B2B

**Comment by in-session** [comment]
I think failing tests are not related to changes from this PR

**Comment by paras89** [comment]
@magento run Functional Tests B2B

**Review by fredden — APPROVED** [review]

**Comment by in-session** [comment]
Follow-up on https://github.com/magento/magento2/pull/37563

**Comment by engcom-Hotel** [comment]
@magento create issue

**Comment by engcom-Bravo** [comment]
Hi @in-session,

Thanks for the collaboration & contribution!

### :heavy_check_mark: QA Passed

**Preconditions:**
- Install fresh Magento **`2.4-develop`** and **`PHP 8.1`**


**Before:** :heavy_multiplication_x: 

**colon is not accepted by the pattern.**

Screenshot from 2023-12-27 13-03-32




**After:** :heavy_check_mark:  

**colon is accepted by the pattern.**

Screenshot from 2023-12-27 15-23-29




Thanks.

File: app/code/Magento/PageBuilder/view/adminhtml/web/js/form/element/validator-rules-mixin.js [hunk 1]
```diff
@@ -67,7 +67,7 @@ define([
      * @return {Boolean}
      */
     function validateCssClass(str) {
-        return (/^[a-zA-Z _\-\d]+$/i).test(str);
+        return (/^[a-zA-Z\d\-_/:.\[\]&amp;@()! ]*$/i).test(str);
```
Comment (fredden) [inline app/code/Magento/PageBuilder/view/adminhtml/web/js/form/element/validator-rules-mixin.js]: I think `&amp;` here should be `&`. This isn't an HTML document, but a JavaScript file. `a` and `m` and `p` are already included in `a-z`, and I think `;` shouldn't be added as a valid class name character.

Edit: also, should we be allowing empty classes here? (Suggest reverting the change from `]+$` to `]*$`.)

```suggestion
        return (/^[a-zA-Z\d\-_/:.\[\]&@()! ]+$/i).test(str);
```

**Review by fredden — CHANGES_REQUESTED** [review]

File: app/code/Magento/PageBuilder/view/adminhtml/web/js/form/element/validator-rules-mixin.js (hunk 1 above)
Comment (in-session) [reply app/code/Magento/PageBuilder/view/adminhtml/web/js/form/element/validator-rules-mixin.js]: Thanks for the tip, you're absolutely right

**Review by ihor-sviziev — APPROVED** [review]

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by engcom-Bravo** [comment]
### :heavy_check_mark: QA Passed

**Preconditions:**
- Install fresh Magento **`2.4-develop`** and **`PHP 8.1`**


**Before:** :heavy_multiplication_x: 

**colon is not accepted by the pattern.**






**After:** :heavy_check_mark:  

**colon is accepted by the pattern.**






Thanks.

**Review by fredden — APPROVED** [review]

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B, Functional Tests CE, Functional Tests EE, Integration Tests, Performance Acceptance Tests, Static Tests, Unit Tests

**Comment by engcom-Hotel** [comment]
We can close this PR as it has already been merged.

**Pull request closed without merging**